### PR TITLE
Feature/Add information about TEA, PR, PRV, Indicators and Prog. Indicators

### PR DESCRIPTION
**Pull request by nshandra**
### References
Issues:
-  [Add information about Tracked Entity Attributes](https://app.clickup.com/t/3jna3h5)
-  [Add information about Program Rules](https://app.clickup.com/t/3jna4dv)
-  [Add information about Program Rule Variables](https://app.clickup.com/t/3jna4j2)
-  [Add information about Indicators](https://app.clickup.com/t/3jna4bf)
-  [Add information about Program Indicators](https://app.clickup.com/t/3jna3zr)

### Description
- Add information about Tracked Entity Attributes
    - Name
    - Form name
    - code
    - description
    - Option set (+ options)
    - Value Type
    - Aggregation Type

- Add information about Indicators
    - Add information about relevant Indicators, defined as those ones that contain Program Indicator or DE of the Program selected. They might be identified also through a manual assignation to an indicator group
    - Name, Description, Numerator, Denominator
    - Stage(s) of reference: filled out based on DE or PI included

- Add information about Program Indicators
    - Name
    - Description
    - Filter: elements formatted as “Stage name : DE name” or “TEI : TEI name”. Option codes formatted according to real option names, not codes
    - Calculation: elements formatted as “Stage name : DE name” or “TEI : TEI name”. Option codes formatted according to real option names, not codes
    - Aggregation Type
    - Type of Program indicator: Event/Enrollment
    - Boundaries
    - Stage(s) of reference: filled out case filter includes the component “V{program_stage_id} ==” or with the stage to which the data elements belong to

- Add information about Program Rules
    - Name
    - Description
    - Program Stage: formatted with name, not id
    - Program Rule Expression: same format considerations as before
    - Program Rule Action: fields according to type of program rule action

- Add information about Program Rule Variables
    - Name
    - Source type
    - Program stage
    - Data element
    - TEI

### NOTES
The use of `$rootScope`  to communicate between controllers is a crutch, because i didn't managed to make the intended method of "Scope Events Propagation" work.

### CODE REVIEW

File: app/dossierPrograms/dossierPrograms.services.js [hunk 1]
```diff
@@ -2,25 +2,20 @@
...
 var qryProgramStageSections =
     dhisUrl +
-    "programStages/:programStageId.json?fields=id,displayName,repeatable,sortOrder,programStageSections[id,displayName,dataElements[displayName,displayFormName,displayDescription,valueType,optionSetValue,optionSet[options[displayName]]]],programStageDataElements[dataElement[displayName,displayFormName,displayDescription,valueType,optionSetValue,optionSet[options[displayName]]]]&paging=false";
-
-var qryProgramIndicators =
-    dhisUrl + "programs/:programId.json?fields=programIndicators[displayName,displayDescription,expression,filter]";
-
-var qryProgramIndicatorExpressions = dhisUrl + "programIndicators/expression/description";
-var qryProgramIndicatorFilters = dhisUrl + "programIndicators/filter/description";
+    "programStages/:programStageId?fields=id,displayName,repeatable,sortOrder,\
```
Comment (tokland) [inline app/dossierPrograms/dossierPrograms.services.js]: [comment] I see that this long string-concatenation styles come from the original code, but now that we have to change it, some proposals:

1) [simple] Use  [field1, field2, ...].join(",") (directly or with a thin wrapper).
2) More complex, a d2-api style wrapper that converts something like `params({id: true, name: true, options: {id: true, code: true}})` to string.  (complex, a probably overkill here).

File: app/dossierPrograms/dossierPrograms.view.html [hunk 2]
```diff
@@ -89,12 +89,15 @@ <h3 class="section" id="{{ section.id }}">{{ section.displayName }}</h3>
...
                                 <td>{{ dataElement.valueType }}</td>
                                 <td>
-                                    <select ng-if="dataElement.optionSet.options" class="form-control col-md-2">
-                                        <option selected>See options</option>
-                                        <option ng-repeat="option in dataElement.optionSet.options">
-                                            {{ option.displayName }}
-                                        </option>
-                                    </select>
+                                    <div ng-if="dataElement.optionSet.options">
+                                        <p>Option Set name: {{ dataElement.optionSet.name }}</p>
```
Comment (tokland) [inline app/dossierPrograms/dossierPrograms.view.html]: Do we need `| translate` here? (and other literals in the file)

File: app/dossierPrograms/dossierPrograms.view.html [hunk 3]
```diff
@@ -103,32 +106,64 @@ <h3 class="section" id="{{ section.id }}">{{ section.displayName }}</h3>
...
+                            <th>{{ "Boundaries" | translate }}</th>
                         </tr>
                     </thead>
                     <tbody>
-                        <tr ng-repeat="indicator in  indicators.programIndicators | orderBy: 'displayName'">
-                            <td>{{indicator.displayName}}</td>
-                            <td>{{indicator.displayDescription}}</td>
-                            <td>{{indicator.expression}}</td>
-                            <td>{{indicator.filter}}</td>
+                        <tr ng-repeat="indicator in  programIndicators | orderBy: 'displayName'">
```
Comment (tokland) [inline app/dossierPrograms/dossierPrograms.view.html]: [very minor, TOC] two spaces

**Review by tokland — COMMENTED** [review]
Just some minor comments, but otherwise code-wise looks very good to me.

File: app/dossierPrograms/dossierPrograms.view.html (hunk 3 above)
Comment (nshandra) [reply app/dossierPrograms/dossierPrograms.view.html]: Done.

File: app/dossierPrograms/dossierPrograms.view.html (hunk 2 above)
Comment (nshandra) [reply app/dossierPrograms/dossierPrograms.view.html]: Yes, an overlook by my part.

File: app/dossierPrograms/dossierPrograms.services.js (hunk 1 above)
Comment (nshandra) [reply app/dossierPrograms/dossierPrograms.services.js]: Switched to the first proposal.

**Review by tokland — APPROVED** [review]

**Pull request merged**